Tune bust XP costs to make them proportional to breast volume

ezsh opened the pull request:
Disclaimer: made just for fun. While parts of this PR could be useful (XP computation, unit system update) the main motivation behind it was a pure fun and I'm quite surprised by the fact that the final outcome appears to be usable. But I wouldn't open a PR if did not want it to be merged, of course.

After the fail of #19 with the "physical" units applied to the bust size, I decided to  that the failure was not due to pulling numbers where they are unneeded, but due to an insufficient amount of the number and formulae behind that attempt. So here is what came out of the second attempt.

I noticed that total XP points required for leveling the bust stat from 0 to the given level may be fitted with a second order polynomial function quite well. From the other side, if we model a breast as an inflating balloon on a basement of a fixed size, its volume will be proportional to the second order of the balloon height, which is in turn proportional to the cup size. A little bit of googling confirmed that real breast follow the same dependence (at least to the sizes measured by medics). OK, then we can add required dynamic to the cup sizes converting the total XP points for the bust size into a volume. 

So, I took the real dependence ([bust.pdf](https://github.com/JJM152/Queen/files/2522154/bust.pdf)) and fitted the bust XP curve:

[bust_level_comparison.pdf](https://github.com/JJM152/Queen/files/2522144/bust_level_comparison.pdf)

assuming that 3 stat levels correspond to a single EU cup size. And a similar plot for leveling costs:
[bust_level_cost_comparison.pdf](https://github.com/JJM152/Queen/files/2522146/bust_level_cost_comparison.pdf)

As you can see, they fit quite well (is is a coincidence?), thus I went ahead and added required functions to display breast volume, to make it possible to define leveling costs as a function, and tuned quest and job requirement so the total required XP change as little as possible.

It was fun to do that, maybe players can get fun using this?

ezsh commented:
Rebased.

JJM152 commented on src/000-SCRIPT_OBJ/Helpers.js:
```diff
@@ -107,23 +107,24 @@ App.PR = new function() {
+	    var lastSmallerProp;
+	    for (var prop in Ratings) {
+	        if (!Ratings.hasOwnProperty(prop)) continue;
+	        if (prop > Value) break;
```
If the rating is 0 there will be no lastSmallerProp and the function returns undefined, but I'm actually defining 0 in some instances for the leveling - it'll never hit, will it?

ezsh replied:
It is the indent change that makes it look like a new function body. But in fact I just added the line 118. As for the question itself, I think it would be simpler if we use `>=` here. But as far as I remember, all the leveling records are written assuming `>`?

JJM152 commented:
It's fine because it doesn't supplant the old leveling system and you did the quest clean up to make it make sense, but just please keep in mind that this was set up intentionally to be easily understood and changed by people who want to build their own version of the game. If somehow the declarative version could be worked into this in a commented code block with a hint to use that instead of the function if they wanted to toy with it that would be the best of both worlds. thanks!

JJM152 commented:
Seems I've managed to put these last few remaining PR's into a state of conflict, sorry about that 😭

ezsh commented:
> If somehow the declarative version could be worked into this in a commented code block with a hint to use that instead of the function if they wanted to toy with it that would be the best of both worlds.

Maybe a note saying this particular stat shows a sample how XP points can be defined by a function, but there may be a table like in other cases will be OK?

ezsh commented:
Rebased and added a brief explanation for the function-type leveling table. 

Also noticed a weird comparison in the `Player.GetLeveling()` inside the loop. I think I overlooked that in #137 and it has to be changed to compare stat value but not percent with the leveling table. The same holds true for the new branch (the functional one).

ezsh commented:
ping?

JJM152 commented:
Just holding off on this until I get a chance to look at how this may impact the game pacing and some quests / requirements.
